rust/connlib/tunnel/src/client.rs

```diff
@@ -576,43 +592,54 @@ impl ClientState {
 
         self.gateways_site.insert(gateway_id, site_id);
         self.recently_connected_gateways.put(gateway_id, ());
+        self.peers
+            .add_ips_with_resource(&gateway_id, &ips, &resource_id);
```

Comment:
@conectado In the future, we might want to consider doing this at the very top, before we check `awaiting_connection_details`. There is no reason why the "routing details" aren't valid and shouldn't be saved, even if we don't have an entry in that map.

Currently, this won't happen because we only ever receive `on_connection_details` in response to a `connection_intent` but I think it could make sense to decouple that.